pkg/transform/oauth/oauth.go

```diff
@@ -107,9 +123,11 @@ func Transform(oauthconfig *configv1.OAuthConfig) (*OAuthCRD, []secrets.Secret,
 			}
 		default:
 			logrus.Infof("Can't handle %s OAuth kind", kind)
+			continue
 		}
 		oauthCrd.Spec.IdentityProviders = append(oauthCrd.Spec.IdentityProviders, idP)
-		if secret != (secrets.Secret{}) {
+
+		if secret.Metadata.Name != "htpasswd_auth-secret" || p.Kind == "HTPasswdPasswordIdentityProvider" {
```

Comment:
Consider setting `htpasswd_auth-secret` and `HTPasswdPasswordIdentityProvider` as constants

Reply:
Hopefully this looks better now.